skimage/future/manual_segmentation.py

```diff
@@ -0,0 +1,216 @@
+from functools import reduce
+import numpy as np
+import matplotlib
+import matplotlib.pyplot as plt
+from matplotlib.patches import Polygon
+from matplotlib.collections import PatchCollection
+from ..draw import polygon
+
+
+LEFT_CLICK = 1
+RIGHT_CLICK = 3
+
+
+def _mask_from_vertices(vertices, shape, label):
+    mask = np.zeros(shape, dtype=int)
+    pr = [y for x, y in vertices]
+    pc = [x for x, y in vertices]
+    rr, cc = polygon(pr, pc, shape)
+    mask[rr, cc] = label
+    return mask
+
+
+def _draw_polygon(ax, vertices, alpha=0.4):
+    polygon = Polygon(vertices, closed=True)
+    p = PatchCollection([polygon], match_original=True, alpha=alpha)
+    polygon_object = ax.add_collection(p)
+    plt.draw()
+    return polygon_object
+
+
+def manual_polygon_segmentation(image, alpha=0.4, return_all=False):
```

Comment:
@soupault you ask @pskeshu to add it to that file, or you merge and then you immediately submit a PR. =)